Include/cpython/objimpl.h

```diff
@@ -43,20 +43,23 @@ PyAPI_FUNC(Py_ssize_t) _PyGC_CollectIfEnabled(void);
      && (Py_TYPE(o)->tp_is_gc == NULL || Py_TYPE(o)->tp_is_gc(o)))
 
 /* GC information is stored BEFORE the object structure. */
-typedef struct {
-    // Pointer to next object in the list.
-    // 0 means the object is not tracked
-    uintptr_t _gc_next;
-
-    // Pointer to previous object in the list.
-    // Lowest two bits are used for flags documented later.
-    uintptr_t _gc_prev;
+typedef union {
+    struct {
+        // Pointer to next object in the list.
+        // 0 means the object is not tracked
+        uintptr_t next;
+
+        // Pointer to previous object in the list.
+        // Lowest two bits are used for flags documented later.
+        uintptr_t prev;
+    } _gc;
+    long double _dummy;  /* for worst alignment */
```

Comment:
Can you please elaborate the comment? Add "bpo-27987" and explain that x64-64 ABI requires to align on 16 bytes.

Do you have to align to 16 bytes on x86 (32-bit) as well?

Reply:
Oh! I forgot about 32bit platform.
I thought this PR does nothing, but it makes PyGC_Head to 16 bytes from 8 bytes on x86 platform.

Maybe, we can use `__attribute__ ((aligned))` if compiler is GCC or clang.